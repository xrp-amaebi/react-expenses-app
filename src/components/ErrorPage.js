import React from 'react';
import {Link} from 'react-router-dom';

const ErrorPage = () => ( 
    <div>
        404 - Error Page Not Found. <Link to="/">Back to DashBoard</Link> 
    </div>
);
export default ErrorPage;