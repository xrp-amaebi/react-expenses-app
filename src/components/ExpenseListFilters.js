import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters';
import { DateRangePicker } from 'react-dates';


export class ExpenseListFilters extends React.Component {
    state = {
        focused: null
    }

    onDatesChange = ({startDate, endDate}) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };

    onFocusChange = focused => {
        this.setState(() => ({ focused }));
    };

    onSortChange = e => { 
        if (e.target.value === 'amount') {
            this.props.sortByAmount();
        } else if(e.target.value === 'date' ) {
            this.props.sortByDate();
        }
    }

    onTextChange = e => this.props.setTextFilter(e.target.value);

    render() {
        return ( 
            <div className={'content-container'}>
                <div className={'input-group'}>
                    <div className={'input-group__item'}>
                        <input
                            className={'text-input'}
                            onChange={this.onTextChange}
                            placeholder={'search expenses'}
                            type={"text"}
                            value={this.props.filters.text}
                        />
                    </div>
                    <div className={'input-group__item'}>
                        <select
                            className={'select'}
                            value={this.props.filters.sortBy}
                            onChange={this.onSortChange}
                        >
                            <option value="date">{'Date'}</option>
                            <option value="amount">{'Amount'}</option>
                        </select>
                    </div>
                    <div className={'input-group__item'}>
                        <DateRangePicker
                            startDate={this.props.filters.startDate}
                            endDate={this.props.filters.endDate}
                            onDatesChange={this.onDatesChange}
                            focusedInput={this.state.focused}
                            onFocusChange={this.onFocusChange}
                            showClearDates={true}
                            numberOfMonths={1}
                            isOutsideRange={() => false}
                        />
                    </div>
                </div>
            </div> 
        );
    }
};

const mapStateToProps = (state) => ({
    filters: state.filters
});

const mapDispatchToProps = (dispatch, props) => ({
    setTextFilter: (target) => dispatch(setTextFilter(target)),
    sortByAmount: () => dispatch(sortByAmount()),
    sortByDate: () => dispatch(sortByDate()),
    setStartDate: (startdate) => dispatch(setStartDate(startdate)),
    setEndDate: (enddate) => dispatch(setEndDate(enddate))
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters); 