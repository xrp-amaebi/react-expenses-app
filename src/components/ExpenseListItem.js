 import React from 'react';
 import {Link} from 'react-router-dom';

 const ExpenseListItems = ({description, amount, createdAt, id}) => (
    <div>
      <Link to={ `/edit/${id}` } activeclassname='is-active'>
         <h3>{description}</h3>
      </Link>
        <p>{amount} - {createdAt}</p>
    </div>
);

export default ExpenseListItems; 