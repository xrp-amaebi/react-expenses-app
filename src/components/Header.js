import React from 'react';
import {NavLink} from 'react-router-dom';


const Header = ()=>( 
    <header>
        <h1>Expensify</h1>

            <NavLink to="/" activeclassname="is-active" exact={true}>Visit DashBoard</NavLink>
            <NavLink to="/create" activeclassname="is-active">Back to Create</NavLink>
            <NavLink to="/help" activeclassname="is-active">Back to Help</NavLink>
    </header>
);
export default Header;