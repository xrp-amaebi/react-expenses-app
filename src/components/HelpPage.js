import React from 'react';

export const HelpPage = () => ( 
    <div>
        This is from my Help component.
    </div>
);
