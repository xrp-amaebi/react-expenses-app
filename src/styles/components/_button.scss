.button
{
    background: $blue;
    border: none;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $small;
    text-decoration: none;
}

.button--link
{   
    background: none;
}   