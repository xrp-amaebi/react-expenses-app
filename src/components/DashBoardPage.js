import React from "react";
import ExpenseList from "./ExpensesList";
import ExpenseListFilters from "./ExpenseListFilters";
import ExpenseSummary from './ExpenseSummary';


export const DashBoard = () => ( 
    <div>
        <ExpenseSummary /> 
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);