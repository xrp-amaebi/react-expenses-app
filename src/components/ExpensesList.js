import React from 'react';
import { connect } from 'react-redux'; 
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';


export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No Expenses</p>) : 
            (
                props.expenses.map(item => 
                    <ExpenseListItem 
                        key={item.id} 
                        {...item} 
                    />
                )
            )
        }   
    </div>
);

const mapStateToProps = (state) => ({ 
    expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpenseList);