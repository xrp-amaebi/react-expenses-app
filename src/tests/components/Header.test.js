import React from 'react'; 
import { Header } from '../../components/Header'; 
import { shallow } from 'enzyme';


test('should render header correctly', () => {
    const wrapper = shallow(<Header  startLogout={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
    // const renderer = new ReactShallowRenderer();
    // renderer.render(<Header/>);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('should call startLogout on buttonClick', () => {
    const startLogout =  jest.fn();
    const wrapper = shallow(<Header startLogout={startLogout} />);
    wrapper.find('button').simulate('click');
    expect(startLogout).toHaveBeenCalled();
}); 