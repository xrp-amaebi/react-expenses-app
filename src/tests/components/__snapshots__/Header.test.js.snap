// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="active"
    activeclassname="is-active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    Visit DashBoard
  </NavLink>
  <NavLink
    activeClassName="active"
    activeclassname="is-active"
    ariaCurrent="true"
    to="/create"
  >
    Back to Create
  </NavLink>
  <NavLink
    activeClassName="active"
    activeclassname="is-active"
    ariaCurrent="true"
    to="/help"
  >
    Back to Help
  </NavLink>
</header>
`;
